widgets/HTTP-streamingsource.json

```diff
@@ -502,5 +497,188 @@
+      "condition": {
+        "property": "paginationType",
+        "operator": "equal to",
+        "value": "Increment an index"
+      },
+      "show": [
+        {
+          "name": "startIndex",
+          "type": "property"
+        },
+        {
+          "name": "maxIndex",
+          "type": "property"
+        },
+        {
+          "name": "indexIncrement",
+          "type": "property"
+        }
+      ]
+    },
+    {
+      "name": "Token in Response Body",
+      "condition": {
+        "property": "paginationType",
+        "operator": "equal to",
+        "value": "Token in response body"
+      },
+      "show": [
+        {
+          "name": "nextPageTokenPath",
+          "type": "property"
+        },
+        {
+          "name": "nextPageUrlParameter",
+          "type": "property"
+        }
+      ]
+    },
+    {
+      "name": "Link in response body",
+      "condition": {
+        "property": "paginationType",
+        "operator": "equal to",
+        "value": "Link in response body"
+      },
+      "show": [
+        {
+          "name": "nextPageFieldPath",
+          "type": "property"
+        }
+      ]
+    },
+    {
+      "name": "Custom pagination",
+      "condition": {
+        "property": "paginationType",
+        "operator": "equal to",
+        "value": "Custom"
+      },
+      "show": [
+        {
+          "name": "customPaginationCode",
+          "type": "property"
+        }
+      ]
+    },
+    {
+      "name": "Pagination none",
+      "condition": {
+        "property": "paginationType",
+        "operator": "equal to",
+        "value": "None"
+      },
+      "show": [
+        {
+          "name": "waitTimeBetweenPages",
+          "type": "property"
+        }
+      ]
+    },
+    {
+      "name": "OAuth 2 disabled",
+      "condition": {
+        "property": "oauth2Enabled",
+        "operator": "equal to",
+        "value": "false"
+      },
+      "show": [
+        {
+          "name": "username",
+          "type": "property"
+        },
+        {
+          "name": "password",
+          "type": "property"
+        }
+      ]
+    },
+    {
+      "name": "OAuth 2 enabled",
+      "condition": {
+        "property": "oauth2Enabled",
+        "operator": "equal to",
+        "value": "true"
+      },
+      "show": [
+        {
+          "name": "authUrl",
+          "type": "property"
+        },
+        {
+          "name": "tokenUrl",
+          "type": "property"
+        },
+        {
+          "name": "clientId",
+          "type": "property"
+        },
+        {
+          "name": "clientSecret",
+          "type": "property"
+        },
+        {
+          "name": "scopes",
+          "type": "property"
+        },
+        {
+          "name": "refreshToken",
+          "type": "property"
+        }
+      ]
+    },
+    {
+      "name": "JSON/XML Formatting",
+      "condition": {
+        "expression": "format == 'json' || format == 'xml'"
```

Comment:
I was going to open a jira about this. Shouldn't we instead support operators such as 'in', 'not in', 'matches', 'not matches', 'contains', 'not contains'? For simple conditions like these, we shouldn't need to specify an expression right?

@elfenheart @albertshau

Reply:
+1. I think the workaround today would be to have two filters, one for json and one for xml.

Reply:
https://issues.cask.co/browse/CDAP-16023

Reply:
This is right now possible. https://github.com/TomFrost/Jexl#a-note-about-in